docs/src/main/asciidoc/deployment-to-openshift.adoc

```diff
@@ -0,0 +1,104 @@
+
+For more information about the Quarkus CLI, see xref:cli-tooling.adoc[Building Quarkus apps with Quarkus CLI].
+
+* By using the Maven or Gradle plugins
++
+You can use the Maven and Gradle plugins to deploy Quarkus applications by running the provided deploy commands.
+For example, if you are using Maven, issue the `mvn quarkus:deploy` command.
+If you are using Gradle, issue the `gradle deploy` command.
+
+Use the information provided here to determine the strategy to use to build and deploy your Quarkus applications on OpenShift.
+
+[#overviewbuildstrategies]
+== Overview of OpenShift Container Platform build strategies
+
+OpenShift binary build:: This strategy leverages the OpenShift container image framework and uses a JAR file as input to the build process, which speeds up building and deploying your application.
+As an out-of-the-box feature, the OpenShift extension is configured to use the `quarkus-container-image-openshift` extension to build the container image inside the OpenShift cluster.
+For information about the OpenShift container image, see xref:container-image.adoc#openshift[container-image-openshift].
+
+non-OpenShift build:: As an alternative to the `quarkus-container-image-openshift` approach, Quarkus also supports the following extensions for building container images.
+
+- xref:container-image.adoc#docker[container-image-docker]
+- xref:container-image.adoc#jib[container-image-jib]
+- xref:container-image.adoc#buildpack[container-image-buildpack]
+
+[NOTE]
+====
+The OpenShift Container Platform Docker build strategy is the preferred non-OpenShift build strategy because it supports Quarkus applications targeted for JVM or compiled to native executables.
+However, for compatibility with earlier Quarkus versions, the default build strategy is the OpenShift binary build.
+To select the OpenShift Container Platform Docker build strategy, use the `quarkus.openshift.build-strategy` property.
+====
+
+If you use a non-OpenShift container image extension, an `ImageStream` is created that points to an external `dockerImageRepository`.
+The image is built and pushed to the registry and the `ImageStream` populates the tags that are available in the `dockerImageRepository`.
+To select one of the above extensions to use to build the image, issue the following command and specify the corresponding `builder` value.
+
+For example, to select the `container-image-docker` extension, issue the following command:
+
+[source,bash]
+----
+quarkus.container-image.builder=docker
+----
+
+The Docker build strategy builds the artifacts outside the OpenShift Container Platform cluster, locally or in a continuous integration (CI) environment, and provides them to the OpenShift Container Platform build system together with a Dockerfile.
+The artifacts include JAR files or a native executable.
+The container gets built inside the OpenShift Container Platform cluster and is provided as an image stream.
+
+== Build strategies supported by Quarkus
```

Comment:
I think that this table is confusing and maybe we are better off without it.

Reply:
Removed table.